erpnext/stock/doctype/serial_no/serial_no.py

```diff
@@ -203,14 +252,15 @@ def auto_fetch_serial_number(
 	for_doctype: Optional[str] = None,
 	exclude_sr_nos=None,
 ) -> List[str]:
+	"""NOTE: Unused but kept for backward compatibility."""
 
 	filters = frappe._dict({"item_code": item_code, "warehouse": warehouse})
 
 	if exclude_sr_nos is None:
 		exclude_sr_nos = []
 	else:
 		exclude_sr_nos = safe_json_loads(exclude_sr_nos)
-		exclude_sr_nos = get_serial_nos(clean_serial_no_string("\n".join(exclude_sr_nos)))
+		exclude_sr_nos = get_serial_nos(clean_serial_no_string("\n".join(exclude_sr_nos), item_code))
```

Comment:
I can't make much sense of this line.

I suspect the original intention was to indirectly call `.upper().strip()` on each serial number. That could also be achieved by the simpler

```suggestion
		exclude_sr_nos = get_serial_nos("\n".join(exclude_sr_nos))
```

, since

```python
serial_numbers = "\n".join([" a", "b "]
(
	get_serial_nos(clean_serial_no_string(serial_numbers))
	== get_serial_nos(serial_numbers)
	== ["A", "B"]
)
```

If we add the `item_code` parameter, the behavior differs in that we'd get `None` for missing serial numbers instead of the cleaned up string.

Reply:
`get_serial_nos` is mainly used to handle legacy data where the serial no text field is used.
Anyway, it will return the serial no back as is if it cannot be found to keep it consistent with older behaviour.
The retrieval of serial no name is added as a defensive approach.